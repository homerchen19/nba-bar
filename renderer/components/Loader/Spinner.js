import React from 'react';
import { Flex } from 'antd-mobile';
import MDSpinner from 'react-md-spinner';

import { colors } from '../../styles/theme';

const Spinner = () => (
  <Flex justify="center" style={{ height: '100vh', width: '100%', flex: 1 }}>
    <MDSpinner
      color1={colors.blue}
      color2={colors.red}
      color3={colors.blue}
      color4={colors.red}
      size={32}
    />
  </Flex>
);

export default Spinner;
