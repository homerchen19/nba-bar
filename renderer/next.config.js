const withCSS = require('@zeit/next-css');
const withImages = require('next-images');

module.exports = withImages(
  withCSS({
    cssModules: true,
    webpack(config, { isServer }) {
      config.target = 'electron-renderer';

      if (isServer) {
        const antStyles = /antd-mobile\/.*?\/style.*?/;
        const origExternals = [...config.externals];
        config.externals = [
          (context, request, callback) => {
            if (request.match(antStyles)) return callback();
            if (typeof origExternals[0] === 'function') {
              origExternals[0](context, request, callback);
            } else {
              callback();
            }
          },
          ...(typeof origExternals[0] === 'function' ? [] : origExternals),
        ];

        config.module.rules.unshift({
          test: antStyles,
          use: 'null-loader',
        });
      }

      return config;
    },
    exportPathMap() {
      return {
        '/home': { page: '/home' },
        '/live': { page: '/live' },
        '/preview': { page: '/preview' },
        '/scoreboard': { page: '/scoreboard' },
        '/settings': { page: '/settings' },
        '/standings': { page: '/standings' },
      };
    },
  })
);
