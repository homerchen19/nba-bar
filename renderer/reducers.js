import { combineReducers } from 'redux';

import homeReducer from './containers/Home/reducer';
import previewReducer from './containers/Preview/reducer';
import liveReducer from './containers/Live/reducer';
import scoreboardReducer from './containers/Scoreboard/reducer';
import standingsReducer from './containers/Standings/reducer';
import settingsReducer from './containers/Settings/reducer';

const rootReducer = combineReducers({
  home: homeReducer,
  preview: previewReducer,
  live: liveReducer,
  scoreboard: scoreboardReducer,
  standings: standingsReducer,
  settings: settingsReducer,
});

export default rootReducer;
