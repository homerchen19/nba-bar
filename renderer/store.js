import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';

import rootReducer from './reducers';

const middleware = [thunk, logger];

const enhancers = compose(
  applyMiddleware(...middleware),
  typeof window !== 'undefined' &&
    window.devToolsExtension &&
    process.env.NODE_ENV !== 'production'
    ? window.devToolsExtension()
    : f => f
);

export default function configureStore(initialState = {}) {
  const store = createStore(rootReducer, initialState, enhancers);

  return store;
}
