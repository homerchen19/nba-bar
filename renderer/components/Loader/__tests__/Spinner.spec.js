import React from 'react';
import renderer from 'react-test-renderer';

import Spinner from '../Spinner';

it('be defined', () => {
  expect(Spinner).toBeDefined();
});

it('match snapshot', () => {
  const tree = renderer.create(<Spinner />).toJSON();

  expect(tree).toMatchSnapshot();
});
