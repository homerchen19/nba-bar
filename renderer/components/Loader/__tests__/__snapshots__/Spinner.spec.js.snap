// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="am-flexbox am-flexbox-justify-center am-flexbox-align-center"
  style={
    Object {
      "flex": 1,
      "height": "100vh",
      "width": "100%",
    }
  }
>
  <span
    style={
      Object {
        "animation": "__react-md-spinner-animation__root-rotate 1568.235294117647ms linear infinite",
        "display": "inline-block",
        "height": 32,
        "position": "relative",
        "verticalAlign": "middle",
        "width": 32,
      }
    }
  >
    <span
      style={
        Object {
          "animationDuration": "5332ms",
          "animationIterationCount": "infinite",
          "animationName": "__react-md-spinner-animation__fill-unfill-rotate, __react-md-spinner-animation__layer-1-fade-in-out",
          "animationTimingFunction": "cubic-bezier(.4, 0, .2, 1)",
          "borderColor": "#0069b9",
          "display": "flex",
          "height": "100%",
          "opacity": 1,
          "position": "absolute",
          "whiteSpace": "nowrap",
          "width": "100%",
        }
      }
    >
      <span
        style={
          Object {
            "borderColor": "inherit",
            "display": "inline-block",
            "flexGrow": 1,
            "height": "100%",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <span
          style={
            Object {
              "animationDuration": "1333ms",
              "animationIterationCount": "infinite",
              "animationName": "__react-md-spinner-animation__left-spin",
              "animationTimingFunction": "cubic-bezier(.4, 0, .2, 1)",
              "borderColor": "#0069b9 transparent transparent #0069b9",
              "borderRadius": "50%",
              "borderStyle": "solid",
              "borderWidth": 3,
              "bottom": 0,
              "boxSizing": "border-box",
              "left": 0,
              "position": "absolute",
              "top": 0,
              "transform": "rotate(129deg)",
              "width": "200%",
            }
          }
        />
      </span>
      <span
        style={
          Object {
            "borderColor": "inherit",
            "display": "inline-block",
            "flexGrow": 1,
            "height": "100%",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <span
          style={
            Object {
              "animationDuration": "1333ms",
              "animationIterationCount": "infinite",
              "animationName": "__react-md-spinner-animation__right-spin",
              "animationTimingFunction": "cubic-bezier(.4, 0, .2, 1)",
              "borderColor": "#0069b9 #0069b9 transparent transparent",
              "borderRadius": "50%",
              "borderStyle": "solid",
              "borderWidth": 3,
              "bottom": 0,
              "boxSizing": "border-box",
              "left": "-100%",
              "position": "absolute",
              "top": 0,
              "transform": "rotate(-129deg)",
              "width": "200%",
            }
          }
        />
      </span>
      <span
        style={
          Object {
            "borderColor": "inherit",
            "borderRadius": "50%",
            "borderTopStyle": "solid",
            "borderWidth": 3,
            "boxSizing": "border-box",
            "left": "45%",
            "position": "absolute",
            "top": 0,
            "width": "10%",
          }
        }
      />
    </span>
    <span
      style={
        Object {
          "animationDuration": "5332ms",
          "animationIterationCount": "infinite",
          "animationName": "__react-md-spinner-animation__fill-unfill-rotate, __react-md-spinner-animation__layer-2-fade-in-out",
          "animationTimingFunction": "cubic-bezier(.4, 0, .2, 1)",
          "borderColor": "#f00b47",
          "display": "flex",
          "height": "100%",
          "opacity": 1,
          "position": "absolute",
          "whiteSpace": "nowrap",
          "width": "100%",
        }
      }
    >
      <span
        style={
          Object {
            "borderColor": "inherit",
            "display": "inline-block",
            "flexGrow": 1,
            "height": "100%",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <span
          style={
            Object {
              "animationDuration": "1333ms",
              "animationIterationCount": "infinite",
              "animationName": "__react-md-spinner-animation__left-spin",
              "animationTimingFunction": "cubic-bezier(.4, 0, .2, 1)",
              "borderColor": "#f00b47 transparent transparent #f00b47",
              "borderRadius": "50%",
              "borderStyle": "solid",
              "borderWidth": 3,
              "bottom": 0,
              "boxSizing": "border-box",
              "left": 0,
              "position": "absolute",
              "top": 0,
              "transform": "rotate(129deg)",
              "width": "200%",
            }
          }
        />
      </span>
      <span
        style={
          Object {
            "borderColor": "inherit",
            "display": "inline-block",
            "flexGrow": 1,
            "height": "100%",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <span
          style={
            Object {
              "animationDuration": "1333ms",
              "animationIterationCount": "infinite",
              "animationName": "__react-md-spinner-animation__right-spin",
              "animationTimingFunction": "cubic-bezier(.4, 0, .2, 1)",
              "borderColor": "#f00b47 #f00b47 transparent transparent",
              "borderRadius": "50%",
              "borderStyle": "solid",
              "borderWidth": 3,
              "bottom": 0,
              "boxSizing": "border-box",
              "left": "-100%",
              "position": "absolute",
              "top": 0,
              "transform": "rotate(-129deg)",
              "width": "200%",
            }
          }
        />
      </span>
      <span
        style={
          Object {
            "borderColor": "inherit",
            "borderRadius": "50%",
            "borderTopStyle": "solid",
            "borderWidth": 3,
            "boxSizing": "border-box",
            "left": "45%",
            "position": "absolute",
            "top": 0,
            "width": "10%",
          }
        }
      />
    </span>
    <span
      style={
        Object {
          "animationDuration": "5332ms",
          "animationIterationCount": "infinite",
          "animationName": "__react-md-spinner-animation__fill-unfill-rotate, __react-md-spinner-animation__layer-3-fade-in-out",
          "animationTimingFunction": "cubic-bezier(.4, 0, .2, 1)",
          "borderColor": "#0069b9",
          "display": "flex",
          "height": "100%",
          "opacity": 1,
          "position": "absolute",
          "whiteSpace": "nowrap",
          "width": "100%",
        }
      }
    >
      <span
        style={
          Object {
            "borderColor": "inherit",
            "display": "inline-block",
            "flexGrow": 1,
            "height": "100%",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <span
          style={
            Object {
              "animationDuration": "1333ms",
              "animationIterationCount": "infinite",
              "animationName": "__react-md-spinner-animation__left-spin",
              "animationTimingFunction": "cubic-bezier(.4, 0, .2, 1)",
              "borderColor": "#0069b9 transparent transparent #0069b9",
              "borderRadius": "50%",
              "borderStyle": "solid",
              "borderWidth": 3,
              "bottom": 0,
              "boxSizing": "border-box",
              "left": 0,
              "position": "absolute",
              "top": 0,
              "transform": "rotate(129deg)",
              "width": "200%",
            }
          }
        />
      </span>
      <span
        style={
          Object {
            "borderColor": "inherit",
            "display": "inline-block",
            "flexGrow": 1,
            "height": "100%",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <span
          style={
            Object {
              "animationDuration": "1333ms",
              "animationIterationCount": "infinite",
              "animationName": "__react-md-spinner-animation__right-spin",
              "animationTimingFunction": "cubic-bezier(.4, 0, .2, 1)",
              "borderColor": "#0069b9 #0069b9 transparent transparent",
              "borderRadius": "50%",
              "borderStyle": "solid",
              "borderWidth": 3,
              "bottom": 0,
              "boxSizing": "border-box",
              "left": "-100%",
              "position": "absolute",
              "top": 0,
              "transform": "rotate(-129deg)",
              "width": "200%",
            }
          }
        />
      </span>
      <span
        style={
          Object {
            "borderColor": "inherit",
            "borderRadius": "50%",
            "borderTopStyle": "solid",
            "borderWidth": 3,
            "boxSizing": "border-box",
            "left": "45%",
            "position": "absolute",
            "top": 0,
            "width": "10%",
          }
        }
      />
    </span>
    <span
      style={
        Object {
          "animationDuration": "5332ms",
          "animationIterationCount": "infinite",
          "animationName": "__react-md-spinner-animation__fill-unfill-rotate, __react-md-spinner-animation__layer-4-fade-in-out",
          "animationTimingFunction": "cubic-bezier(.4, 0, .2, 1)",
          "borderColor": "#f00b47",
          "display": "flex",
          "height": "100%",
          "opacity": 1,
          "position": "absolute",
          "whiteSpace": "nowrap",
          "width": "100%",
        }
      }
    >
      <span
        style={
          Object {
            "borderColor": "inherit",
            "display": "inline-block",
            "flexGrow": 1,
            "height": "100%",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <span
          style={
            Object {
              "animationDuration": "1333ms",
              "animationIterationCount": "infinite",
              "animationName": "__react-md-spinner-animation__left-spin",
              "animationTimingFunction": "cubic-bezier(.4, 0, .2, 1)",
              "borderColor": "#f00b47 transparent transparent #f00b47",
              "borderRadius": "50%",
              "borderStyle": "solid",
              "borderWidth": 3,
              "bottom": 0,
              "boxSizing": "border-box",
              "left": 0,
              "position": "absolute",
              "top": 0,
              "transform": "rotate(129deg)",
              "width": "200%",
            }
          }
        />
      </span>
      <span
        style={
          Object {
            "borderColor": "inherit",
            "display": "inline-block",
            "flexGrow": 1,
            "height": "100%",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <span
          style={
            Object {
              "animationDuration": "1333ms",
              "animationIterationCount": "infinite",
              "animationName": "__react-md-spinner-animation__right-spin",
              "animationTimingFunction": "cubic-bezier(.4, 0, .2, 1)",
              "borderColor": "#f00b47 #f00b47 transparent transparent",
              "borderRadius": "50%",
              "borderStyle": "solid",
              "borderWidth": 3,
              "bottom": 0,
              "boxSizing": "border-box",
              "left": "-100%",
              "position": "absolute",
              "top": 0,
              "transform": "rotate(-129deg)",
              "width": "200%",
            }
          }
        />
      </span>
      <span
        style={
          Object {
            "borderColor": "inherit",
            "borderRadius": "50%",
            "borderTopStyle": "solid",
            "borderWidth": 3,
            "boxSizing": "border-box",
            "left": "45%",
            "position": "absolute",
            "top": 0,
            "width": "10%",
          }
        }
      />
    </span>
  </span>
</div>
`;
